modules/mobile/app/controllers/mobile/v0/appointments_controller.rb

```diff
@@ -66,15 +68,25 @@ def clear_appointments_cache
       end
 
       def fetch_appointments(validated_params)
-        appointments = appointments_cache_interface.fetch_appointments(
+        appointments, failures = appointments_cache_interface.fetch_appointments(
           user: @current_user, start_date: validated_params[:start_date], end_date: validated_params[:end_date],
           fetch_cache: validated_params[:use_cache]
         )
 
         appointments.filter! { |appt| appt.is_pending == false } unless include_pending?(validated_params)
         appointments.reverse! if validated_params[:reverse_sort]
 
-        appointments
+        [appointments, failures]
+      end
+
+      def partial_errors(failures)
```

Comment:
I would suggest adding a comment above this method explaining what partial errors are and how they work in the mobile app.

Reply:
added!